dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java

```diff
@@ -2256,11 +2341,51 @@ private void buildCase(CaseTree tree, int index) {
           scan(stmt, null);
         }
       } else {
-        scan(TreeUtils.caseTreeGetBody(tree), null);
+        Tree bodyTree = TreeUtils.caseTreeGetBody(tree);
+        if (bodyTree instanceof ExpressionTree) {
+          buildSwitchExpressionResult((ExpressionTree) bodyTree);
+        } else {
+          scan(bodyTree, null);
+        }
       }
       extendWithExtendedNode(new UnconditionalJump(nextBodyL));
       addLabelForNextNode(nextCaseL);
     }
+
+    /**
+     * Does the following for each result expression of a switch expression:
```

Comment:
This comment seems wrong.  The client has the logic that this sentence described.  This method only handles one result expression at a time.